modules/eventlog/src/main/java/org/apache/ignite/internal/eventlog/ser/JacksonBasedJsonSerializer.java

```diff
@@ -18,26 +18,26 @@
 package org.apache.ignite.internal.eventlog.ser;
 
 import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.SerializerProvider;
 import com.fasterxml.jackson.databind.module.SimpleModule;
 import com.fasterxml.jackson.databind.ser.std.StdSerializer;
 import java.io.IOException;
 import org.apache.ignite.internal.eventlog.api.Event;
-import org.apache.ignite.internal.eventlog.event.EventImpl;
 import org.apache.ignite.internal.eventlog.event.EventUser;
 import org.apache.ignite.internal.lang.IgniteInternalException;
 import org.apache.ignite.lang.ErrorGroups.Common;
 
-/** Serializes events to JSON. */
-public class JsonEventSerializer implements EventSerializer {
+/** Serializes events to JSON. Uses provided json serializer to serialize events of specified class. */
+public class JacksonBasedJsonSerializer<T extends Event> implements EventSerializer {
     private final ObjectMapper mapper;
 
     /** Default constructor. */
-    public JsonEventSerializer() {
+    public JacksonBasedJsonSerializer(Class<T> eventClass, JsonSerializer<T> eventJsonSerializer) {
         mapper = new ObjectMapper();
         SimpleModule module = new SimpleModule();
-        module.addSerializer(EventImpl.class, new EventImplJacksonSerializer());
+        module.addSerializer(eventClass, eventJsonSerializer);
         module.addSerializer(EventUser.class, new EventUserJacksonSerializer());
```

Comment:
Can you please define such a registry that can be used to register a serializer for any type, including EventUser? I don't like that we can register the serializer for the Event but can not register a serializer for EventUser with the same approach.